examples/learned_models/edmd_utils.jl

```diff
@@ -0,0 +1,226 @@
+abstract type AbstractController end
+
+resetcontroller!(::AbstractController, x0) = nothing
+
+struct RandomController{D} <: AbstractController
+    m::Int
+    distribution::D
+    function RandomController(model::RD.AbstractModel, 
+            distribution::D=Normal()) where D 
+        new{D}(RD.control_dim(model), distribution)
+    end
+end
+
+function getcontrol(ctrl::RandomController, x, t)
+    u = rand(ctrl.distribution, ctrl.m)
+    return u
+end
+
+"""
+Picks a random control value each time the controller is reset, 
+and applies it as a constant.
+"""
+struct RandConstController{D} <: AbstractController
+    distribution::D
+    u::Vector{Float64}
+    function RandConstController(distribution::D) where D
+        u = rand(distribution)
+        new{D}(distribution, u)
+    end
+end
+
+resetcontroller!(ctrl::RandConstController, x0) = rand!(ctrl.distribution, ctrl.u)
+
+getcontrol(ctrl::RandConstController, x, t) = ctrl.u
+
+struct ZeroController <: AbstractController 
+    m::Int
+    ZeroController(model::RD.AbstractModel) = new(RD.control_dim(model))
+end
+getcontrol(ctrl::ZeroController, x, t) = zeros(ctrl.m)
+
+struct LQRController <: AbstractController
+    K::Matrix{Float64}
+    xeq::Vector{Float64}
+    ueq::Vector{Float64}
+end
+
+function LQRController(A,B,Q,R, xeq, ueq)
+    K = dlqr(A,B,Q,R)
+    LQRController(Matrix(K), Vector(xeq), Vector(ueq))
+end
+
+function LQRController(model::RD.DiscreteDynamics, Q, R, xeq, ueq, dt)
+    n,m = RD.dims(model)
+    n != length(xeq) && throw(DimensionMismatch("Expected a state vector of length $n, got $(length(xeq))."))
+    m != length(ueq) && throw(DimensionMismatch("Expected a control vector of length $m, got $(length(ueq))."))
+    zeq = RD.KnotPoint{n,m}(xeq, ueq, 0.0, dt)
+    J = zeros(n, n+m)
+    xn = zeros(n)
+    RD.jacobian!(RD.default_signature(model), RD.default_diffmethod(model), model, J, xn, zeq)
+    A = J[:,1:n]
+    B = J[:,n+1:n+m]
+    LQRController(A,B,Q,R,xeq,ueq)
+end
+
+function dlqr(A,B,Q,R; max_iters=200, tol=1e-6)
+    P = Matrix(copy(Q))
+    n,m = size(B)
+    K = zeros(m,n)
+    K_prev = copy(K)
+    
+    for k = 1:max_iters
+        K .= (R + B'P*B) \ (B'P*A)
+        P .= Q + A'P*A - A'P*B*K
+        if norm(K-K_prev,Inf) < tol
+            println("Converged in $k iterations")
+            return K
+        end
+        K_prev .= K
+    end
+    return K * NaN
+end
+
+function getcontrol(ctrl::LQRController, x, t)
+    dx = x - ctrl.xeq
+    ctrl.ueq - ctrl.K*dx
+end
+
+struct TVLQRController <: AbstractController
+    K::Vector{Matrix{Float64}}
+    xref::Vector{Vector{Float64}}
+    uref::Vector{Vector{Float64}}
+    time::Vector{Float64}
+end
+
+get_k(ctrl::TVLQRController, t) = min(searchsortedfirst(ctrl.time, t), length(ctrl.K))
+
+function getcontrol(ctrl::TVLQRController, x, t)
+    k = get_k(ctrl, t)
+    K = ctrl.K[k]
+    dx = x - ctrl.xref[k]
+    ctrl.uref[k] + K*dx
+end
+
+mutable struct ALTROController{D} <: AbstractController
```

Comment:
Added this controller. Made mutable so that the `prob` field can be easily updated.